HomeWork2/Nataliia_Kuchynska_HW2_optional.cpp

```diff
@@ -0,0 +1,143 @@
+#include <iostream>
+#include <limits>
+
+const int MAX_VAL = std::numeric_limits<int>::max();
+const int MINBITSET = 0;
+const int MAXBITSET = 32;
+void CristmasTree() {
+  std::cout << " Please enter the number in the range from 0 to 100"
+            << std::endl;
+  int n;
+  do {
+    std::cin >> n;
+    if (n > 0 && n < 101) {
+      break;
+    }
+    std::cout << " Incorrect value. Try again." << std::endl;
+  } while (true);
+  if (n % 2 == 0) {
+    n += 1;
+  }
+  const int n_half = n / 2;
+  for (int i = 0; i < n_half + 1; ++i) {
+    for (int j = 0; j < n; j++) {
+      if (j < (n_half - i) || j > (n_half + i)) {
+        std::cout << " ";
+      } else {
+        std::cout << "*";
+      }
+    }
+    std::cout << std::endl;
+  }
+}
+void BitCounter() {
+  std::cout << " Please enter the number in the range from 0 to "
+            << std::numeric_limits<short>::max() << std::endl;
+  int n;
+  do {
+    std::cin >> n;
+    if (n > 0 && n < std::numeric_limits<short>::max()) {
+      break;
+    }
+    std::cout << "Incorrect value. Try again." << std::endl;
+  } while (true);
+  unsigned bit_counter = 0;
+  while (n) {
```

Comment:
Я рекомендую условия писать всегда явно
```suggestion
  while (n != 0) {
```